### PR TITLE
re2: fix for pkgsStatic

### PR DESCRIPTION
**Pull request by tobim**
Not using `lib.optional` to avoid changing the `.drv` hash.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/re2/default.nix [hunk 1]
```diff
@@ -17,10 +17,14 @@ stdenv.mkDerivation {
     substituteInPlace Makefile  --replace "SED_INPLACE=sed -i '''" "SED_INPLACE=sed -i"
   '';
 
+  buildFlags = if stdenv.hostPlatform.isStatic then [ "static" ] else null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/re2/default.nix:20]:
```suggestion
  buildFlags = lib.optionals stdenv.hostPlatform.isStatic [ "static" ];;
```

File: pkgs/development/libraries/re2/default.nix [hunk 2]
```diff
@@ -17,10 +17,14 @@ stdenv.mkDerivation {
...
+
   preCheck = "patchShebangs runtests";
   doCheck = true;
   checkTarget = "test";
 
+  installTargets = if stdenv.hostPlatform.isStatic then [ "static-install" ] else null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/re2/default.nix:26]:
```suggestion
  installTargets = lib.optionals stdenv.hostPlatform.isStatic [ "static-install" ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
> Not using `lib.optional` to avoid changing the `.drv` hash.

This is causing 26 rebuilds. Please do it properly.

**Comment by tobim** [comment]
> This is causing 26 rebuilds.

ofBorg disagrees.

**Comment by SuperSandro2000** [comment]
> ofBorg disagrees.

Using lib.optional does 26 rebuilds which is totally fine to merge into master. Please only use rebuild avoiding hacks when it would cause stdenv rebuilds. This is not such change so we don't need to do it.

**Comment by tobim** [comment]
@SuperSandro2000 mind having another look?

**Comment by SuperSandro2000** [comment]
Checked it manually and it builds.

**Pull request merged**